smalltalksrc/VMMakerTests/VMPrimitiveTest.class.st

```diff
@@ -1599,74 +1793,365 @@ VMPrimitiveTest >> testPrimitiveGreaterThanWithNoErrorPopsOperands [
+	
+	interpreter push: array2.
+	
+	interpreter primitiveArrayBecome.
+	
+	interpreter push: object1. 
+	interpreter push: array2. 
+	
+	interpreter primitiveIdentical. 
 
-	interpreter push: object.
-	interpreter push: (memory integerObjectOf: 1).
-	interpreter push: memory falseObject.
-	interpreter primitiveInstVarAtPut.
+	self assert: interpreter primFailCode equals: PrimErrBadArgument.
 	
-	self deny: interpreter failed.
-	self assert: (memory fetchPointer: 0 ofObject: object) equals: memory falseObject.
 ]
 
-{ #category : #'tests - primitiveLessOrEqual' }
-VMPrimitiveTest >> testPrimitiveLessOrEqualFailsWithFloat [
+{ #category : #'tests - primitiveIdentical' }
+VMPrimitiveTest >> testPrimitiveIdenticalFailsForForwardedOopOnOtherObject [
+	|object1|
+	"Tests the case where objectArg > 1 and: isOopFowarded: stackTop"
+	
+	object1 := self setUpForwardedObjects.
+	
+	interpreter argumentCount: 3.
+	
+	interpreter push: (memory integerObjectOf: 16). 
+	interpreter push: object1.
+	 
+	interpreter primitiveIdentical. 
 
-	|classFloat|
+	self assert: interpreter failed. 
+	self assert: interpreter primFailCode equals: PrimErrBadArgument.
+	
+]
 
-	classFloat := self newClassInOldSpaceWithSlots: 0 instSpec: memory firstLongFormat.
-	memory setHashBitsOf: classFloat to: ClassFloatCompactIndex.
-	memory
-		storePointer: ClassFloatCompactIndex
-		ofObject: memory classTableFirstPage
-		withValue: classFloat.
-		
-	"This simulated classFloat class is necessary because the 32bits VM cannot instanciate boxed floats by itself"
+{ #category : #'tests - primitiveIdentical' }
+VMPrimitiveTest >> testPrimitiveIdenticalFailsForForwardedOopOnReceiver [
+	|object1|
 	
-	interpreter push: (memory integerObjectOf: 16).
-	interpreter push: (memory floatObjectOf: 16.0).
+	object1 := self setUpForwardedObjects. 
 	
-	interpreter primitiveLessOrEqual.
+	interpreter push: object1. 
+	interpreter push: (memory integerObjectOf: 16). 
+	
+	interpreter primitiveIdentical. 
+
+	self assert: interpreter failed. 
+	self assert: interpreter primFailCode equals: PrimErrBadArgument.
+	
+]
+
+{ #category : #'tests - primitiveIdentical' }
+VMPrimitiveTest >> testPrimitiveIdenticalReturnsBool [
+
+	interpreter push: memory trueObject.
+	interpreter push: (memory integerObjectOf: 2).
+	
+	interpreter primitiveIdentical. 
+
+	self assert: interpreter stackTop equals: memory falseObject. 
+	
+]
+
+{ #category : #'tests - primitiveIdentical' }
+VMPrimitiveTest >> testPrimitiveIdenticalReturnsFalseForObjectsWithSameValueButDifferentType [
+	| object1 object2 |
+	
+	object1:= memory integerObjectOf: 16. 
+	object2:= memory floatObjectOf: 16.0 . 
+	
+	interpreter push: object1 . 
+	interpreter push: object2. 
+	
+	interpreter primitiveIdentical. 
+
+	self assert: interpreter stackTop equals: memory falseObject. 
+	
+]
+
+{ #category : #'tests - primitiveIdentical' }
+VMPrimitiveTest >> testPrimitiveIdenticalReturnsFalseWithWrongArgCount [
+	| object |
```

Comment:
This test is a bit strange. I'd assume you are accessing things that are not accessible in the stack...